src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+
+Base.:*(d::NonMatrixDistribution, σ::Real) = σ * d
+Base.:/(d::NonMatrixDistribution, τ::Real) = Base.inv(τ) * d
+Base.:\(τ::ArrayLike, d::NonMatrixDistribution) = Base.inv(τ) * d
+
+
+#### Extremes
+
+## Univariate
+
+function maximum(d::UnivariateAffine)
+    maxim = d.σ > 0 ? maximum(d.ρ) : minimum(d.ρ)
+    return d.μ + d.σ * maxim
+end
+
+function minimum(d::UnivariateAffine)
+    minim = d.σ > 0 ? minimum(d.ρ) : maximum(d.ρ)
+    return d.μ + d.σ * minim
+end
+
+function extrema(d::UnivariateAffine)
+    extremes = extrema(d.ρ)
+    extremes = _flip(d.σ, extremes)
+    return @. d.μ + d.σ * extremes
+end
+
+
+## Multivariate
+# when finding the extrema, we need to flip the extremes if the scale is negative
+_flip(signed::Real, tuple::Tuple) = signed > 0 ? tuple : (last(tuple), first(tuple))
+
+function minimum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    # The minimum of d.ρ[i] will minimize d[j] if d.σ[i,j] is negative. 
+    # Otherwise the maximum will minimize d[j]
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    return mins
+end
+
+function maximum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return maxes
+end
+
+function extrema(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return (mins, maxes)
+end
+
+support(d::AffineDistribution) = affine_support(d.μ, d.σ, support(d.ρ))
+affine_support(μ::Real, σ::Real, support) = μ .+ σ .* support
+
+function affine_support(μ::Real, σ::Real, support::RealInterval) 
+    if σ > 0
+        lower = support.lb
+        upper = support.ub
+    else
+        lower = support.ub
+        upper = support.lb
+    end
+    return RealInterval(μ + σ * lower, μ + σ * upper)
+end
+
+
+#### Conversions
+
+function convert(::Type{AffineDistribution{F,S,Tμ,Tσ,D}}, d::AffineDistribution) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    return AffineDistribution(Tμ(d.μ), Tσ(d.σ), convert(D, d.ρ))
+end
+
+
+function Base.eltype(::Type{<:AffineDistribution{F,S,Tμ,Tσ,D}}) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    T = Core.Compiler.return_type(Base.:*, (Tσ, eltype(D)))
+    return Core.Compiler.return_type(Base.:+, (Tμ, T))
```

Comment:
We should not use `return_type` (generally, usually one shouldn't). It is not guaranteed to return concrete types and e.g. may return `Any` at any point.

Moreover, the general opinion around `eltype` is that it should only be affected by the parameters if absolutely necessary. E.g., for continuous distributions it should not be affected by the parameters.

Reply:
The main problem is that `eltype` actually is affected by the parameters, if `eltype(ρ)` and `typeof(σ)` are different, and it's not clear to me how else to deal with that.